File: caesar.py
#!/usr/bin/python3

import argparse

parser = argparse.ArgumentParser(
        description="Caesar Encryptor/Decryptor.")

parser.add_argument("-dec",dest="mode",action="store_true",help="For decryption.")

args = parser.parse_args()

if args.mode == 0:
    text = input("Message to be encrypted: ")
    key = int(input("Key(Integer): ")) #luam cheia de tip integer
    encrypted = ""
    for char in text:
        if char != ' ':
            encrypted = encrypted + chr(ord(char)+key) #transformam caracterul in integer (ASCII) 
        else:                                           #si adunam cheia dupa care transformam inapoi in caracter
            encrypted = encrypted + ' '
    print ("Encrypted text is: {}".format(encrypted))
else:
    encrypted = input("Encrypted message to be decrypted: ")
    key = int(input("Key(Integer): "))
    text = ""
    for char in encrypted:
        if char != ' ':
            text = text + chr(ord(char)-key)            #transformam caracterul in integer (ASCII)
        else:                                           #si scadem cheia dupa care transformam inapoi in caracter
            encrypted = encrypted + ' '
    print ("Decrypted text is: {}".format(text))

File: caesar.cpp

#include <iostream>
using namespace std;

int main(){
    string c;
    char n[50];
    int k;
    char mod;
    cout << "What do you want to do enc(e)/dec(d): ";
    cin >> mod;
    if (mod == 'e'){
        cout << "Enter a message to be encrypted: ";
        cin.ignore();
        getline(cin, c);
        cout << "Enter Key: ";
        cin >> k;
        for (int i=0; i < c.size();i++){
            if (c[i] != ' '){
                n[i] = char(int(c[i]) + k); // transformam caracterul in cod ASCII, rezultatul il adunam cu cheia si il transformam inapoi in caracter
            }
            else{
                n[i] += ' ';     // In caz ca este spatiu il lasam neschimbat (in caz ca vrem sa encriptam si spatiul scoatem if-ul
            }
    }
    cout << "Encrypted message: " << n << endl;
    }
    else{
        cout << "Enter a message to be decrypted: ";
        cin.ignore();
        getline(cin, c);
        cout << "Enter Key: ";
        cin >> k;
        for (int i = 0; i < c.size();i++){
            if (c[i] != ' '){
                n[i] = char(int(c[i]) - k);// transformam caracterul in cod ASCII, rezultatul il scadem cu cheia si il transformam inapoi in caracter
            }
            else{
                n[i] += ' ';// In caz ca este spatiu il lasam neschimbat (in caz ca vrem sa encriptam si spatiul scoatem if-ul
            }
        }
        cout << "Decrypted message: " << n << endl;
    }
    return 0;
}
